ironfish-cli/src/commands/chain/import.ts

```diff
@@ -0,0 +1,161 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { DEFAULT_SNAPSHOT_BUCKET_URL, FileUtils } from '@ironfish/sdk'
+import { CliUx, Flags } from '@oclif/core'
+import axios from 'axios'
+import crypto from 'crypto'
+import fsAsync from 'fs/promises'
+import os from 'os'
+import path from 'path'
+import tar from 'tar'
+import { v4 as uuid } from 'uuid'
+import { IronfishCommand } from '../../command'
+import { RemoteFlags } from '../../flags'
+import { ProgressBar } from '../../types'
+
+export default class ImportSnapshot extends IronfishCommand {
+  static hidden = false
+
+  static description = `Import chain snapshot`
+
+  static flags = {
+    ...RemoteFlags,
+    bucket: Flags.string({
+      char: 'b',
+      parse: (input: string) => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Bucket URL to download snapshot from',
+    }),
+    path: Flags.string({
+      char: 'p',
+      parse: (input: string): Promise<string> => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Path to snapshot file',
+    }),
+    confirm: Flags.boolean({
+      default: false,
+      description: 'confirm without asking',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(ImportSnapshot)
+
+    let snapshotPath
+    const tempDir = path.join(os.tmpdir(), uuid())
+    await fsAsync.mkdir(tempDir, { recursive: true })
+
+    if (flags.path) {
+      snapshotPath = flags.path
+    } else {
+      const bucket = (flags.bucket || DEFAULT_SNAPSHOT_BUCKET_URL || '').trim()
+      if (!bucket) {
+        this.log(`Cannot download snapshot without bucket URL`)
+      }
+
+      const client = await this.sdk.connectRpc()
+      const status = await client.getChainInfo()
+
+      const manifest = await axios
+        .get<{
+          checksum: string
+          file_name: string
+          file_size: number
+          timestamp: number
+          block_height: number
+        }>(`https://${bucket}/manifest.json`)
```

Comment:
It'd be nice to extract the type here to another file, and share it with the `snapshot` command, so it at least avoids any potential consistency issues between creating the manifest file and parsing it.